extensions/spring-cloud-config-client-oidc/deployment/src/main/java/io/quarkus/spring/cloud/config/client/SpringCloudConfigClientOidcProcessor.java

```diff
@@ -0,0 +1,23 @@
+package io.quarkus.spring.cloud.config.client;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.deployment.Feature;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+
+public class SpringCloudConfigClientOidcProcessor {
+
+    @BuildStep
+    public void feature(BuildProducer<FeatureBuildItem> feature) {
+        feature.produce(new FeatureBuildItem(Feature.SPRING_CLOUD_CONFIG_CLIENT_OIDC));
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    public void configure(BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
+        additionalBeans.produce(AdditionalBeanBuildItem.unremovableOf(SpringCloudConfigClientOidcProcessor.class));
```

Comment:
I think you mean `SpringCloudClientBasicAuthProvider` here, right?

Reply:
Actually I think you've added it later so I think this method is entirely redundant

Reply:
I should rename the package to add oidc but this is the step that generated the OIDc provider. In the other module, I generate the basicAuth provider.